src/Services/QueuedJobService.php

```diff
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
+            $retryFalloffMultiplier = (float) Config::inst()->get($jobClass, 'retry_falloff_multiplier');
+            $retryFalloffMultiplierVariance = (float) Config::inst()->get(
+                $jobClass,
+                'retry_falloff_multiplier_variance'
+            );
+
+            // Sanitise values so we don't have to deal with edge cases
+            $initialRetryDelay = max(0, $initialRetryDelay);
+            $retryFalloffMultiplier = max(1, $retryFalloffMultiplier);
+            $retryFalloffMultiplierVariance = max(0, $retryFalloffMultiplierVariance);
+
+            $jobConfigs[$jobClass] = [
+                $maxRetryAttempts,
+                $initialRetryDelay,
+                $retryFalloffMultiplier,
+                $retryFalloffMultiplierVariance,
+            ];
+        }
+
+        // We couldn't find any job candidates to retry
+        if (!$jobConfigs) {
+            return;
+        }
+
+        $retryJobIDs = [];
+        $remainingLimit = $jobRetryLimit;
+
+        foreach ($jobConfigs as $jobClass => $retryConfig) {
+            // We've reached the number of jobs we are allowed to retry in this health check run
+            if ($remainingLimit <= 0) {
+                break;
+            }
+
+            [
+                $maxRetryAttempts,
+            ] = $retryConfig;
+
+            $jobIDs = $autoRetryJobs
+                ->filter([
+                    'Implementation' => $jobClass,
+                    'RetryCount:LessThan' => $maxRetryAttempts,
+                ])
+                ->limit($remainingLimit)
+                ->column('ID');
+
+            // Add the found jobs to our list so we can retry these jobs
+            $retryJobIDs = array_merge($retryJobIDs, $jobIDs);
+
+            // Subtract found jobs from the remaining limit
+            $remainingLimit -= count($jobIDs);
+        }
+
+        // We couldn't find any jobs to retry
+        if (!$retryJobIDs) {
+            return;
+        }
+
+        $jobsToRetry = $autoRetryJobs->byIDs($retryJobIDs);
+
+        /** @var QueuedJobDescriptor $jobDescriptor */
+        foreach ($jobsToRetry as $jobDescriptor) {
+            $jobClass = $jobDescriptor->Implementation;
+            $retryConfig = array_key_exists($jobClass, $jobConfigs)
+                ? $jobConfigs[$jobClass]
+                : [];
+
+            // Couldn't find a retry config for this job (possible edge case in parallel processing)
+            if (!$retryConfig) {
+                continue;
+            }
+
+            [
+                $maxRetryAttempts,
+                $initialRetryDelay,
+                $retryFalloffMultiplier,
+                $retryFalloffMultiplierVariance,
+            ] = $retryConfig;
+
+            // Determine the config key that is currently active
+            $retryCount = max(0, $jobDescriptor->RetryCount);
+
+            // We have reached the maximum retry attempts for this job
```

Comment:
```suggestion
            // We have reached the maximum retry attempts for this job (possibly retried in a parallel process after the initial query),
            // so it should remain in its current state
```
just making it abundantly clear why we're skipping it and how we might get into this state given the query in the IDs loop already filters for this.